bamboolean/lexer.py

```diff
@@ -105,64 +108,50 @@ def _integer(self):
             self.next()
         return int(result)
 
+    def skip_n_chars(self, n):
+        for i in range(n):
+            self.next()
+
+    def is_token_equal(self, expected):
+        return expected == reduce(
+            lambda actual, _: actual + str(self.peek()),
+            [self.current_char] * len(expected),
```

Comment:
I don't understand at all what happens here. Perhaps it could be checked more directly?